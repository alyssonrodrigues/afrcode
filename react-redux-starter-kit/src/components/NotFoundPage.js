import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => {
  // TODO refinar página não encontrada
  return (
    <div>
      <h4>
        404 Página não encontrada
      </h4>
      <Link to="/">Retornar</Link>
    </div>
  );
};

export default NotFoundPage;
