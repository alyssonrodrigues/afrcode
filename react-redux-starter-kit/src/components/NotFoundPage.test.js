import React from 'react'
import { shallow } from 'enzyme'

import { NotFoundPage } from './NotFoundPage'

describe('NotFoundPage', () => {
  it('NotFoundPage render deveria ocorrer sem falhas!', () => {
    const componentWrapper = shallow(
      <NotFoundPage />
    )
    expect(componentWrapper).toMatchSnapshot()
  })
})
