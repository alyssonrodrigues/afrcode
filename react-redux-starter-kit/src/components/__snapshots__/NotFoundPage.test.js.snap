// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage NotFoundPage render deveria ocorrer sem falhas! 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h4>
          404 Página não encontrada
</h4>,
        <Link
          replace={false}
          to="/"
>
          Retornar
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404 Página não encontrada",
        },
        "ref": null,
        "rendered": "404 Página não encontrada",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Retornar",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Retornar",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h4>
            404 Página não encontrada
</h4>,
          <Link
            replace={false}
            to="/"
>
            Retornar
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404 Página não encontrada",
          },
          "ref": null,
          "rendered": "404 Página não encontrada",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Retornar",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Retornar",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
